Source/DafnyCore/Compilers/Dafny/Compiler-dafny.cs

```diff
@@ -0,0 +1,750 @@
+      // TrStmtList(body, bodyWr);
+      // return lowWr;
+      throw new UnsupportedFeatureException(Token.NoToken, Feature.RunAllTests);
+    }
+
+    protected override ConcreteSyntaxTree CreateWhileLoop(out ConcreteSyntaxTree guardWriter, ConcreteSyntaxTree wr) {
+      throw new UnsupportedFeatureException(Token.NoToken, Feature.RunAllTests);
+    }
+
+    protected override ConcreteSyntaxTree CreateForLoop(string indexVar, string bound, ConcreteSyntaxTree wr, string start = null) {
+      throw new UnsupportedFeatureException(Token.NoToken, Feature.RunAllTests);
+    }
+
+    protected override ConcreteSyntaxTree CreateDoublingForLoop(string indexVar, int start, ConcreteSyntaxTree wr) {
+      throw new UnsupportedFeatureException(Token.NoToken, Feature.RunAllTests);
+    }
+
+    protected override void EmitIncrementVar(string varName, ConcreteSyntaxTree wr) {
+      throw new UnsupportedFeatureException(Token.NoToken, Feature.RunAllTests);
+    }
+
+    protected override void EmitDecrementVar(string varName, ConcreteSyntaxTree wr) {
+      throw new UnsupportedFeatureException(Token.NoToken, Feature.RunAllTests);
+    }
+
+    protected override string GetQuantifierName(string bvType) {
+      throw new UnsupportedFeatureException(Token.NoToken, Feature.Quantifiers);
+    }
+
+    protected override ConcreteSyntaxTree CreateForeachLoop(string tmpVarName, Type collectionElementType, IToken tok,
+      out ConcreteSyntaxTree collectionWriter, ConcreteSyntaxTree wr) {
+      throw new UnsupportedFeatureException(Token.NoToken, Feature.RunAllTests);
+    }
+
+    protected override void EmitDowncastVariableAssignment(string boundVarName, Type boundVarType, string tmpVarName,
+      Type collectionElementType, bool introduceBoundVar, IToken tok, ConcreteSyntaxTree wr) {
+      throw new UnsupportedFeatureException(Token.NoToken, Feature.RunAllTests);
+    }
+
+    protected override ConcreteSyntaxTree CreateForeachIngredientLoop(string boundVarName, int L, string tupleTypeArgs,
+        out ConcreteSyntaxTree collectionWriter, ConcreteSyntaxTree wr) {
+      throw new UnsupportedFeatureException(Token.NoToken, Feature.RunAllTests);
+    }
+
+    protected override void EmitNew(Type type, IToken tok, CallStmt initCall, ConcreteSyntaxTree wr, ConcreteSyntaxTree wStmts) {
+      throw new UnsupportedFeatureException(Token.NoToken, Feature.RunAllTests);
+    }
+
+    protected override void EmitNewArray(Type elementType, IToken tok, List<string> dimensions,
+      bool mustInitialize, [CanBeNull] string exampleElement, ConcreteSyntaxTree wr, ConcreteSyntaxTree wStmts) {
+      throw new UnsupportedFeatureException(Token.NoToken, Feature.RunAllTests);
+    }
+
+    protected override void EmitLiteralExpr(ConcreteSyntaxTree wr, LiteralExpr e) {
+      wr.Write(e.ToString());
+    }
+
+    protected override void EmitStringLiteral(string str, bool isVerbatim, ConcreteSyntaxTree wr) {
+      wr.Write($"\"{str}\"");
```

Comment:
I think you need an escaping mechanism there down the road, for now I suggest you say it's unsupported.
```suggestion
      if(str.Contains("\"") || str.Contains("\\") || !str.All(char.IsAscii)) {
        throw new UnsupportedFeatureException(Token.NoToken, Feature.UnicodeChars);
      } else {
        wr.Write($"\"{str}\"");
      }
```

Reply:
I'm confused, I did test it on programs that use unicode and it worked.